plugins/Diagnostics/Diagnostic/SystemFormatCheck.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+namespace Piwik\Plugins\Diagnostics\Diagnostic;
+
+use Piwik\Config;
+use Piwik\SettingsServer;
+use Piwik\Translation\Translator;
+
+/**
+ * Check that the memory limit is enough.
+ */
+class SystemFormatCheck implements Diagnostic
+{
+    /**
+     * @var Translator
+     */
+    private $translator;
+
+    /**
+     * @var int
+     */
+
+    public function __construct(Translator $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    public function execute()
+    {
+        $label = $this->translator->translate('Installation_SystemCheckProcessorFormat');
```

Comment:
Actually I don't think that `PHP_INT_SIZE` allows to draw conclusions about the system or the processor.
At least on windows it should be possible to install a x86 PHP version on a Windows version that actually is x64. Not sure if it would be possible to compile a x86 version on a linux system that is x64, actually never tried that...
Also I had something in mind that the BCMatch extension might also allow higher numbers even on a x86 system, but nt fully sure about that right now.
So maybe we should only mention that the PHP version is doesn't support 64bit numbers or something like that? 🤔